site/app/templates/officeHoursQueue/SearchStudentQueue.twig

```diff
@@ -0,0 +1,90 @@
+<div id='student_queue_history'>
+    <br>
+    <h2>Student's Queue History</h2>
+    <p>Input Student RCSID:</p>
+    <div id='search-student-queue'>
+        <input id='search-student-queue-input' data-testid='search-student-queue-input' type='text' placeholder="Enter student's name">
+        <button id='search-student-queue-btn' data-testid='search-student-queue-btn' class="btn btn-primary">Search</button>
+    </div>
+    <table id='student-queue-table' class="table table-striped" style="margin-top: 15px;">
+        <caption style="padding: 5px; border: 1px solid var(--standard-light-gray); font-weight: 600;"></caption>
+        <thead>
+            <tr>
+                <th scope='col'>#</th>
+                <th scope='col'>Current state</th>
+                <th scope='col'>Queue</th>
+                <th scope='col'>Time Entered</th>
+                <th scope='col'>Time In Queue</th>
+                <th scope='col'>Time Helped</th>
+                <th scope='col'>Help Started By</th>
+                <th scope='col'>Removal Type</th>
+            </tr>
+        </thead>
+        <tbody id="search-student-tbody">
+
+        </tbody>
+    </table>
+</div>
+
+<script>
+    const search_student_queue = document.getElementById('search-student-queue-input');
+    const search_student_queue_btn = document.getElementById('search-student-queue-btn');
+    const table_body = document.getElementById('search-student-tbody');
+    const table_caption = document.querySelector('#student-queue-table caption')
+    const times_helped_row = document.createElement('tr');
+    const times_helped_cell = document.createElement('td');
+    times_helped_cell.setAttribute('colspan', '8');
+    times_helped_cell.style.backgroundColor = 'var(--standard-vibrant-green)';
+    times_helped_row.appendChild(times_helped_cell);
+    const url = buildCourseUrl(['queue', 'student_search']);
+    search_student_queue_btn.addEventListener('click', event => {
+        table_body.innerHTML = '';
```

Comment:
Using `innerHTML` poses a security risk because user input can be parsed as HTML, leading to potential HTML injection and harmful script execution. Use `textContent` or `text()` instead of `html()` or `innerHTML` to prevent this.